disk/src/com/treode/disk/edit/DriveGroup.scala

```diff
@@ -264,20 +265,21 @@ private class DriveGroup (
         }
       }
 
-      if (errors.hasErrors) {
-        // If there are errors, close files that have been opened, do not
-        // finalize changes.
-        for (drive <- newAttaches) {
-          drive.close()
-        }
-        scheduler.pass (cb, errors)
-      } else {
-        // Otherwise, we can merge our new changes into the queued changes.
-        this.dno = dno
-        attaches :::= newAttaches
-        drains :::= newDrains
-        changers ::= cb
-        queue.engage()
+      val errorlist = errors.result
+
+      errors.result match {
+        case Errors (list) =>
```

Comment:
if you say `case notify @ Errors (_) =>` here, you can use notify for the callback below, and you can remove the `errorlist = errors.result` above